lib/dor/models/editable.rb

```diff
@@ -232,16 +232,51 @@ def use_license=(use_license_machine)
     # @return [String] A description of the rights defined in the default object rights datastream. Can be 'Stanford', 'World', 'Dark' or 'None'
     def default_rights
       xml = defaultObjectRights.ng_xml
-      if xml.search('//rightsMetadata/access[@type=\'read\']/machine/group').length == 1
-        'Stanford'
-      elsif xml.search('//rightsMetadata/access[@type=\'read\']/machine/world').length == 1
-        'World'
-      elsif xml.search('//rightsMetadata/access[@type=\'discover\']/machine/none').length == 1
-        'Dark'
+      machine_read_access = xml.search('//rightsMetadata/access[@type="read"]/machine')
+      machine_discover_access = xml.search('//rightsMetadata/access[@type="discover"]/machine')
+
+      machine_read_access_node = machine_read_access.length == 1 ? machine_read_access.first : nil
+      machine_discover_access_node = machine_discover_access.length == 1 ? machine_discover_access.first : nil
+
+      if machine_read_access_node && machine_read_access_node.search('./group[text()="Stanford" or text()="stanford"]').length == 1
+        if machine_read_access_node.search('./group[@rule="no-download"]').length == 1
+          'stanford-nd'
+        else
+          'stanford'
+        end
+      elsif machine_read_access_node && machine_read_access_node.search('./world').length == 1
+        if machine_read_access_node.search('./world[@rule="no-download"]').length == 1
+          'world-nd'
+        else
+          'world'
+        end
+      elsif machine_read_access_node && machine_read_access_node.search('./location[text()="spec"]').length == 1
+        'loc:spec'
+      elsif machine_read_access_node && machine_read_access_node.search('./location[text()="music"]').length == 1
+        'loc:music'
+      elsif machine_discover_access_node && machine_discover_access_node.search('./world').length == 1
+        'none'
+      elsif machine_discover_access_node && machine_discover_access_node.search('./none').length == 1
+        'dark'
```

Comment:
So, this reads a little funny to me - can you explain what's happening in lines 257-260, @jmartin-sul ?